### DISCUSSION TITLE
Why immediately dismiss when Tree Based navigation with `navigationDestination(item: destination:)`

### DISCUSSION THREAD

**Discussion by MaraMincho** [pointfreeco/swift-composable-architecture#3264] [Q&A] [answered]
Hello, I am working with TCA and utilizing TreeBased Navigation for navigation. I am wrapping the RootView with a `NavigationStack` and using the `.navigationDestination(item:)` modifier to switch screens. I have confirmed that calling `navigationDestination` works correctly to push screens. The store parameter used in navigationDestination is created through the parent reducer's ifLet.

However, the problem arises when I set the view with the NavigationStack as the destination and pass the Reducer as an init parameter. In this case, the dismiss action is passed to the parent reducer, causing the screen to disappear immediately upon appearing. The strange thing is that if I send an action to the reducer that triggers ifLet again, the navigation works correctly.

Could you please help me understand what the issue might be?

![Simulator Screen Recording - iPhone 15 Pro - 2024-07-31 at 17 12 54](https://github.com/user-attachments/assets/a921fc55-e259-4b1c-869a-59b1e62fdf97)

[Sample project is here](https://github.com/MaraMincho/TCAScrollViewBug)

**Comment by mbrandonw** [comment, marked as answer]
Hi @MaraMincho, you have nested `NavigationStack`s in this code sample, and that is not a valid thing to do, even in vanilla SwiftUI. 

For example, The `RootNavigationView` has a stack at its root, and then it can navigate to `ChildNavigationView`, which also has a stack. You will need to refactor this so that there is only one single `NavigationStack`.

Further, the `navigationDestination(item:)` API in SwiftUI is quite buggy when it comes to multiple levels of navigation. You typically cannot deep link more than 1 level.

**Reply by MaraMincho** [reply]
Thank you. I will refactor the code to have a single NavigationStack as you suggested.
